### ISSUE TITLE
Add remote-data testing

### ISSUE THREAD

**Issue by bsipocz** [astropy/pyvo#330]
Currently, the remote-data testing is not run on CI, there should be at least one job that has the option `--remote-data=any` to make sure we're not skipping a few tests.

However, now that the remote-data directives are added to the docstrings, enabling it would mean that all those examples have to be in fact fixed, e.g. having the right output strings, etc., so closing this issue would need a bit of cleanup work.

I'm labeling it as bug, as some of the examples in the narrative docs might in fact contain bugs.

(two docs are being picked up currently, but the rest should be also been checked that all code examples are prepended with `>>>` and `...`, they are picked up for testing)

```
FAILED docs/index.rst::index.rst
FAILED docs/dal/index.rst::index.rst
```

Also, consider using pytest-timeout and pytest-rerunfailues to make the remote-data CI job more robust, we're about to try them in astroquery: https://github.com/astropy/astroquery/issues/2426 and https://github.com/astropy/astroquery/issues/2424

**Comment by bsipocz** [comment]
This should be addressed to a level for v1.4 as quite a lot of tests are failing with the `--remote-data=any` option (and that is without opting the docs into being properly tested.). E.g. as of today I see these:


```
============================================================= short test summary info ==============================================================
FAILED pyvo/dal/tests/test_sia2_remote.py::TestSIACadc::test_service - pyvo.dal.exceptions.DALServiceError: No working capabilities endpoint prov...
FAILED pyvo/dal/tests/test_sia2_remote.py::TestSIACadc::test_datalink_batch - pyvo.dal.exceptions.DALServiceError: No working capabilities endpoi...
FAILED pyvo/dal/tests/test_sia2_remote.py::TestSIACadc::test_pos - pyvo.dal.exceptions.DALServiceError: No working capabilities endpoint provided
FAILED pyvo/dal/tests/test_sia2_remote.py::TestSIACadc::test_band - pyvo.dal.exceptions.DALServiceError: No working capabilities endpoint provided
FAILED pyvo/dal/tests/test_sia2_remote.py::TestSIACadc::test_time - pyvo.dal.exceptions.DALServiceError: No working capabilities endpoint provided
FAILED pyvo/dal/tests/test_sia2_remote.py::TestSIACadc::test_pol - pyvo.dal.exceptions.DALServiceError: No working capabilities endpoint provided
FAILED pyvo/dal/tests/test_sia2_remote.py::TestSIACadc::test_fov - pyvo.dal.exceptions.DALServiceError: No working capabilities endpoint provided
FAILED pyvo/dal/tests/test_sia2_remote.py::TestSIACadc::test_spatial_res - pyvo.dal.exceptions.DALServiceError: No working capabilities endpoint ...
FAILED pyvo/dal/tests/test_sia2_remote.py::TestSIACadc::test_spec_resp - pyvo.dal.exceptions.DALServiceError: No working capabilities endpoint pr...
FAILED pyvo/dal/tests/test_sia2_remote.py::TestSIACadc::test_exptime - pyvo.dal.exceptions.DALServiceError: No working capabilities endpoint prov...
FAILED pyvo/dal/tests/test_sia2_remote.py::TestSIACadc::test_timeres - pyvo.dal.exceptions.DALServiceError: No working capabilities endpoint prov...
FAILED pyvo/dal/tests/test_sia2_remote.py::TestSIACadc::test_publisher_did - pyvo.dal.exceptions.DALServiceError: No working capabilities endpoin...
FAILED pyvo/dal/tests/test_sia2_remote.py::TestSIACadc::test_facility - pyvo.dal.exceptions.DALServiceError: No working capabilities endpoint pro...
FAILED pyvo/dal/tests/test_sia2_remote.py::TestSIACadc::test_collection - pyvo.dal.exceptions.DALServiceError: No working capabilities endpoint p...
FAILED pyvo/dal/tests/test_sia2_remote.py::TestSIACadc::test_instrument - pyvo.dal.exceptions.DALServiceError: No working capabilities endpoint p...
FAILED pyvo/dal/tests/test_sia2_remote.py::TestSIACadc::test_dataproduct_type - pyvo.dal.exceptions.DALServiceError: No working capabilities endp...
FAILED pyvo/dal/tests/test_sia2_remote.py::TestSIACadc::test_target_name - pyvo.dal.exceptions.DALServiceError: No working capabilities endpoint ...
FAILED pyvo/dal/tests/test_sia2_remote.py::TestSIACadc::test_res_format - pyvo.dal.exceptions.DALServiceError: No working capabilities endpoint p...
FAILED pyvo/registry/tests/test_regtap.py::TestGetTables::test_get_tables_table_instance - AssertionError: assert 'This data co...bscore table.' ...
======================================== 19 failed, 271 passed, 1 xfailed, 3 xpassed in 2712.67s (0:45:12) =========================================

```


cc @andamian @tomdonaldson

**Comment by bsipocz** [comment]
(Looking at the logs more carefully, it looks like at least some (if not all) uses a CADC URL for the testing, and CADC is down during the weekend. However, independently of that, this should be picked up by CI as genuine issues could lay low if not tested against an actual service.
